### PR TITLE
Multiple match collections, explicit depth weights

### PR DESCRIPTION
**Pull request by djkapner**
2 things here.
## First thing, multiple point match collections
Instead of just this:
```
"pointmatch": {
  "name": "my_collection",
  ...
}
```
you can now do this:
```
"pointmatch": {
  "name": [
    "my_collection_1",
    "my_collection_2",
    ...
   ],
  ...
}
```
It can still accept non-list entries. But, now I have marshmallow doing a pre-load and converting single non-list entries into a list with one entry. This applies to any `db_params` object. A consequence of this is that if you try to use the input parameters after they have run through the argschema parser, the single entry will have been converted into a list. For stand-alone calls, like the Allen workflow, this does not matter at all. If you're calling the aligner inside of other code, this might come up.
With this input:
```
params = {
  ...
  "pointmatch": {
    "name": "my_collection:",
    ...}
  ...}
```
this will not work:
```
solver = EMaligner.EMaligner(input_data=params, args=[])
assert params['pointmatch']['name'] == 'my_collection'
```
nor will this:
```
solver = EMaligner.EMaligner(input_data=params, args=[])
assert solver.args['pointmatch']['name'] == 'my_collection'
```
These will work:
```
solver = EMaligner.EMaligner(input_data=params, args=[])
assert solver.args['pointmatch']['name'] == ['my_collection']
```
or
```
solver = EMaligner.EMaligner(input_data=copy.deepcopy(params), args=[])
assert params['pointmatch']['name'] == 'my_collection'
```
## Second thing, explicit depth weights:
now, you can do something like this to override other depth settings:
```
  "matrix_assembly": {
    "depth": [0, 1, 2, 3, 100],
    "explicit_weight_by_depth": [1.0, 0.25, 0.167, 0.125, 1000],
```

### CODE REVIEW

File: integration_tests/test_rough.py [hunk 1]
```diff
@@ -5,11 +5,19 @@
...
 from EMaligner import EMaligner
 from EMaligner.utils import EMalignerException
+from marshmallow import ValidationError
 import json
 import os
+import copy
 
-FILE_ROUGH_TILES = './integration_tests/test_files/rough_input_tiles.json'
-FILE_ROUGH_PMS = './integration_tests/test_files/rough_input_matches.json'
+FILE_ROUGH_TILES = \
```
Comment (RussTorres) [inline integration_tests/test_rough.py]: Can these (and the ones in test_montage) use `os.path` or `pathlib` and be relative to `__file__` to help portability?

**Comment by RussTorres** [comment]
I forget the exact context, but I remember thinking there was something weird with how argschema/render-modules uses `pre_load` and `pre_dump` that we should think about for these kind of cases (causing modified input dictionaries).  I am fine with it in this PR, but it has bothered me in testing.

File: EMaligner/utils.py [hunk 2]
```diff
@@ -185,33 +186,40 @@ def get_tileids_and_tforms(stack, tform_name, zvals, fullsize=False, order=2):
...
-        matches = np.array(matches)
+            for name in collection['name']:
+                mlist.append(
+                        renderapi.pointmatch.get_matches_from_group_to_group(
+                            name,
+                            iId,
+                            jId,
+                            owner=collection['owner'],
+                            render=dbconnection))
+        matches = np.concatenate(mlist)
```
Comment (RussTorres) [inline EMaligner/utils.py]: It looks like these used to return lists but now return numpy arrays.  Any reason not to use list operations like extend here?

Comment (djkapner) [reply EMaligner/utils.py]: It looks like no, no reason. I suppose lists would be more in line with renderapi get_matches...()

**Comment by mlnichols** [comment]
Looks good from a quick check, I'm curious about the speed of the double list comprehension in EMaligner.py vs the one for loop though. While more pythonic, I *think* but haven't run the tests yet, that the one for loop and iterating through matches once is a bit quicker

**Comment by djkapner** [comment]
figure_1-5
I was wondering about that also...
Maybe a middle ground of slicing `np.array(list)` is ok

**Comment by RussTorres** [comment]
@dank I think @mlnichols was talking about the replacement of
```python
    pids = []
    qids = []
    for m in matches:
        pids.append(m['pId'])
        qids.append(m['qId'])
    pids = numpy.array(pids)
    qids = numpy.array(pids)
```

believe it or not doing two list comprehensions actually seems faster here when tested on fairly large lists.

**Comment by djkapner** [comment]
ah, i misunderstood.

**Comment by djkapner** [comment]
@mlnichols is this an approve?

**Comment by mlnichols** [comment]
Yep, I approve. Sorry had holidays with friends in town of late.

**Pull request merged**